### ISSUE TITLE
Test Cleanup

### ISSUE THREAD

**Issue by marcharper** [Axelrod-Python/Axelrod#1287]
Our test files are inconsistent with imports. For example to use the `Match` class in tests, various files use any one of

```
import axelrod as axl
match = axl.Match(...)
```
```
import axelrod
match = axelrod.Match(...)
```
```
from axelrod import Match
match = Match(...)
```

I believe the first is our general preference. It would be nice to update the test files and any others to be consistent.

**Comment by drvinceknight** [comment]
> I believe the first is our general preference.

I agree :+1:

**Comment by sudarshan-parvatikar** [comment]
I'm working on this and I had a few questions.
In case of multiple imports as in `test_actions.py`, which has
```
from axelrod import Action
from axelrod.action import UnknownActionError, actions_to_str, str_to_actions
```
Should the `from axelrod.action import ...` import be left as it is, or should it be refactored too?
if yes, how should the syntax look like after refactoring, for this example?

**Comment by drvinceknight** [comment]
I'd prefer it to be refactored but I don't think we have a "convention" yet for this particular example. Perhaps:

```
import axelrod.actions as axl_actions
```

?

**Comment by sudarshan-parvatikar** [comment]
So, just to be clear, this is fine
```
import axelrod.actions as axl_actions
```
Then, wherever ` UnknownActionError, actions_to_str, str_to_actions` are used, it'd be replaced as:
`self.assertRaises(UnknownActionError, Action.from_char, "c")` --> `self.assertRaises(axl_actions.UnknownActionError, Action.from_char, "c")`

?

**Comment by drvinceknight** [comment]
That's what I'm suggesting yes but I wouldn't be surprised if through discussion and review we come up with something much nicer... Essentially I'm saying, that's what I would go with but it might need to be changed.

**Comment by meatballs** [comment]
How about just using the general preference and its resulting namespace? e.g.

```
import axelrod as axl

self.assertRaises(axl.action.UnknownActionError, Action.from_char, "c") 
```

It's one fewer characters in the assertion, fewer import lines and consistent throughout.

**Comment by sudarshan-parvatikar** [comment]
> How about just using the general preference and its resulting namespace? e.g.
> 
> ```
> import axelrod as axl
> 
> self.assertRaises(axl.action.UnknownActionError, Action.from_char, "c") 
> ```
> 
> It's one fewer characters in the assertion, fewer import lines and consistent throughout.

I think this is more simpler than having 2 `import package as xyz` syntax.
So, using @meatballs method is far more feasible. If that's fine, I can continue updating it.

**Comment by drvinceknight** [comment]
Yup, @meatballs's suggestion is better :+1: (you might need to tweak an `__init__.py` file here or there).

**Comment by marcharper** [comment]
IMO it's also fine in this case to use

```
from axelrod.action import Action, UnknownActionError, actions_to_str, str_to_actions
```

particularly if any one of the imported values is frequently used. FYI in Python, a.b.c incurs module lookups (dictionary lookups) each time it's invoked (since the module or it's values could be dynamically overwritten). This is why people sometimes fix a reference outside of a loop:

```
my_function = a.b.c
for p in players:
    a.b.c(p)
```
This can be a surprising speed up in some cases.

**Comment by drvinceknight** [comment]
I'm fine with that as well :+1:

**Comment by sudarshan-parvatikar** [comment]
I am not that proficient in python, so I'll refactor it the way you think will be efficient, since at the end of the day, the code maintenance should be easy. 😅

**Comment by marcharper** [comment]
Sure, also I had a typo in the example, just fixed it

**Comment by sudarshan-parvatikar** [comment]
@marcharper Great! . To summerize, I'll get started on refactor as :
1. update 'import axelrod as axl' in all files.
2.  leave `from axelrod.xxx import yyy, zzz` as it is.

**Comment by marcharper** [comment]
great, thanks!

**Comment by sudarshan-parvatikar** [comment]
I refactored the code, but on running `python -m unittest discover` I get the following error:

```
======================================================================
ERROR: tests.strategies.test_adaptor (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.strategies.test_adaptor
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "/home/tensai/axle/Axelrod/axelrod/tests/strategies/test_adaptor.py", line 5, in <module>
    import axelrod as axl
ImportError: No module named axelrod


----------------------------------------------------------------------
Ran 90 tests in 0.003s

```
I tried some solutions SO, etc., but I can't seem to fix it. (This article)[https://comeroutewithme.com/2018/02/24/python-relative-imports-and-unittests-derp/] explains this problem, but I can't seem to get it. Can anyone give me some pointers?

**Comment by marcharper** [comment]
Hi, I think it's either an installation issue or the tests are being run from the wrong directory. The output on the PR from Appveyor gives different (more expected) test failures.

Try running ./test in the axelrod directory, which should run the test commands without having to install your modified library.

```
======================================================================
ERROR: test_strategies_with_countermeasures_return_their_countermeasures (axelrod.tests.unit.test_strategy_utils.TestInspectStrategy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 80, in test_strategies_with_countermeasures_return_their_countermeasures
    d_geller = axelrod.GellerDefector()
NameError: name 'axelrod' is not defined
======================================================================
ERROR: test_strategies_without_countermeasures_return_their_strategy (axelrod.tests.unit.test_strategy_utils.TestInspectStrategy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 68, in test_strategies_without_countermeasures_return_their_strategy
    tft = axelrod.TitForTat()
NameError: name 'axelrod' is not defined
======================================================================
ERROR: test_cooperator (axelrod.tests.unit.test_strategy_utils.TestLookAhead)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 112, in test_cooperator
    tft = axelrod.Cooperator()
NameError: name 'axelrod' is not defined
======================================================================
ERROR: test_tit_for_tat (axelrod.tests.unit.test_strategy_utils.TestLookAhead)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 119, in test_tit_for_tat
    tft = axelrod.TitForTat()
NameError: name 'axelrod' is not defined
======================================================================
ERROR: test_tft_reacts_to_cooperation (axelrod.tests.unit.test_strategy_utils.TestSimulateMatch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 90, in test_tft_reacts_to_cooperation
    tft = axelrod.TitForTat()
NameError: name 'axelrod' is not defined
======================================================================
ERROR: test_tft_reacts_to_defection (axelrod.tests.unit.test_strategy_utils.TestSimulateMatch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\axelrod\axelrod\tests\unit\test_strategy_utils.py", line 98, in test_tft_reacts_to_defection
    tft = axelrod.TitForTat()
NameError: name 'axelrod' is not defined
----------------------------------------------------------------------
Ran 4473 tests in 303.785s
FAILED (errors=8, expected failures=1)
Starting new match
```

**Comment by drvinceknight** [comment]
Closed by #1308
